qiskit/circuit/quantumcircuit.py

```diff
@@ -465,6 +466,8 @@ def append(self, instruction, qargs=None, cargs=None):
         if not isinstance(instruction, Instruction) and hasattr(instruction, 'to_instruction'):
             instruction = instruction.to_instruction()
 
+        instruction.label = label
```

Comment:
Can you add to the docstring that this label will be added to the user's `Instruction` instance?